### ISSUE TITLE
AuthState grows on serialisation/deserialisation 

### ISSUE THREAD

**Issue by zoltanf** [openid/AppAuth-Android#224]
I'm not sure if I'm doing anything wrong, but when I save and restore the AuthState to/from json the object and the serialisation output grows. I have tracked it down to the serialisation problem in the "mLastTokenResponse" > "additionalParameters" node.

Example:
Step 1:
```
authState.jsonSerializeString().length();
> Result: 7780
```

Step 2:
```
String json = authState.jsonSerializeString();
AuthState.jsonDeserialize(json).jsonSerializeString().length();
> Result: 10348
```

Here is the (relevant part) of serialised json:
![image](https://cloud.githubusercontent.com/assets/1464667/26201012/2c1e2356-3bd1-11e7-9824-2ad68daceb94.png)

And after deserialisation and again serialisation, the relevant part of json2:
![image](https://cloud.githubusercontent.com/assets/1464667/26201066/6cd28400-3bd1-11e7-8541-4c1a0eb097ff.png)

As you can see on every serialisation under "additionalParameters" there is one level of the serialised object added bellow. Initially, before the first serialisation/deserialisation of the AuthState object, this node ("mLastTokenResponse" > "additionalParameters") is {} (empty object).

After some serialisation/deserialisation cycles (every time we refresh the tokens we need to store the serialised AuthState object) this json grows and (eventually) causes an out-of-memory error (on our test device this reached around 100mb after around 20 cycles).

Thank you and if further information is needed please let me know.

**Comment by iainmcgin** [comment]
Whoops, sorry about that. I'll fix it and release a 0.6.1 for just this.

**Comment by zoltanf** [comment]
Thank you!
